pkg/core/interop_system.go

```diff
@@ -272,7 +275,14 @@ func runtimeNotify(ic *interop.Context, v *vm.VM) error {
 
 // runtimeLog logs the message passed.
 func runtimeLog(ic *interop.Context, v *vm.VM) error {
-	msg := fmt.Sprintf("%q", v.Estack().Pop().Bytes())
+	state := v.Estack().Pop().Bytes()
+	if len(state) > MaxNotificationSize {
+		return fmt.Errorf("message length shouldn't exceed %v", MaxNotificationSize)
+	}
+	if !utf8.Valid(state) {
```

Comment:
This one is a part of #1198, probably.

Reply:
Updated commit description and PR comment.